synapse/third_party_rules/access_rules.py

```diff
@@ -717,3 +750,28 @@ def _is_invite_from_threepid(invite: EventBase, threepid_invite_token: str) -> b
         )
 
         return token == threepid_invite_token
+
+    def _user_is_invited_to_room(
+        self, user_id: str, state_events: StateMap[EventBase]
+    ) -> bool:
+        """Checks whether a given user has been invited to a room
+
+        A user has an invite for a room if its state contains a `m.room.member`
+        event with membership type "invite" and their user ID as the state key.
```

Comment:
```suggestion
        event with membership "invite" and their user ID as the state key.
```

There's no such thing as a membership type in the Matrix spec afaik, it's just a membership.